sematic/db/queries.py

```diff
@@ -200,6 +200,19 @@ def get_resources_by_root_id(root_run_id: str) -> List[ExternalResource]:
         return list(set(r[0] for r in results))
 
 
+def get_run_ids_for_resource(external_resource_id: str) -> List[str]:
+    with db().get_session() as session:
+        results = (
+            session.query(Run.id)
+            .filter(ExternalResource.id == external_resource_id)
+            .filter(ExternalResource.id == RunExternalResource.resource_id)
+            .filter(Run.id == RunExternalResource.run_id)
```

Comment:
You don't really need to query `runs` if all you want is a list of run IDs.
```python
session.query(RunExternalResource.run_id)
.filter(ExternalResource.id == external_resource_id)
.filter(ExternalResource.id == RunExternalResource.resource_id)
```
should be sufficient.

Reply:
Good catch!